stan/math/rev/core/chainablestack.hpp

```diff
@@ -11,6 +11,23 @@ class chainable_alloc;
 
 typedef AutodiffStackSingleton<vari, chainable_alloc> ChainableStack;
 
+/**
+ * Instantiates an instance of the ChainableStack if not already
+ * initialized. This function must be called before any autodiff
+ * variable get's instantiated within any thread which performs
+ * reverse mode autodiff operations.
+ */
+/*
+static inline ChainableStack::AutodiffStackStorage* init() {
+#ifdef STAN_THREADS
+  return ChainableStack::init();
+  if (ChainableStack::instance_ == nullptr)
```

Comment:
Does this `if` statement actually get called if the line before it is a `return`?